### PR TITLE
Update CHANGELOG with 0.6.0 changes.

### PR DESCRIPTION
**Pull request by carlosalberto**
(Wondering whether it feels like an overkill to list the related PRs)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,17 +7,32 @@ the release.
 
 ## Unreleased
 
+## v0.6.0 (07-01-2020)
+
+New:
+
+- Add CODEOWNERS to support area owners ([#661](https://github.com/open-telemetry/opentelemetry-specification/pull/661))
+- Add experimental folder for new features ([#632](https://github.com/open-telemetry/opentelemetry-specification/pull/632))
```
Comment (Oberon00) [inline CHANGELOG.md:15]: I don't think these are relevant for users / SIG maintainers reading the spec changelog.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,17 +7,32 @@ the release.
...
+  Most importantly:
   * Configuration should be stored not per Tracer but in the TracerProvider.
   * Active spans are not per Tracer.
-- Add semantic conventions for process resource.
-- Add peer.service to provide a user-configured name for a remote service
+- Do not set any value in Context upon failed extraction ([#671](https://github.com/open-telemetry/opentelemetry-specification/pull/671))
+- Clarify semantic conventions around span start and end time ([#592](https://github.com/open-telemetry/opentelemetry-specification/pull/592))
+
+Removals:
+- Remove zPagesProcessor mentioned in the SDK ([#683](https://github.com/open-telemetry/opentelemetry-specification/pull/683))
```
Comment (Oberon00) [inline CHANGELOG.md:35]: This was also just an editorial cleanup, nothing was really, semantically removed. I would leave that out.

If we put every single PR in the changelog, then one might just as well read the commit log.

**Comment by carlosalberto** [comment]
Thanks for the review @Oberon00, lets go with your suggestions.

**Comment by carlosalberto** [comment]
Merging without approvals to expedite the 0.6.0 release (as many contributors will be on holidays for the next days, and also because this feels like a relatively non-polemical set of changes).

(There's also #656 to be solved, hopefully for 0.7.0 )

**Pull request merged**